engine/src/flutter/impeller/renderer/backend/vulkan/swapchain/swapchain_vk.h

```diff
@@ -16,10 +16,15 @@
 
 #if FML_OS_ANDROID
 #include "impeller/toolkit/android/native_window.h"
+#include "impeller/toolkit/android/surface_transaction.h"
 #endif  // FML_OS_ANDROID
 
 namespace impeller {
 
+#if FML_OS_ANDROID
+using CreateTransactionCB = std::function<android::SurfaceTransaction()>;
```

Comment:
I hate that this uses a callback but I had a bad time looking for some common place to put an interface to be shared between the platform and impeller code anywhere.

ideally we'd have something fairly simple like:

```
class TransactionFactory {

  android::SurfaceTransaction CreateTransaction();
  
  enum TransactionMode {
    kNative,
    kJava,
  }
  
  void SetTransactionMode(/* java or native*/ mode)
}
```

But I plan to fix this in a follow up.